Sources/ServiceLifecycle/ServiceGroup.swift

```diff
@@ -106,6 +108,37 @@ public actor ServiceGroup: Sendable, Service {
         self.maximumCancellationDuration = configuration._maximumCancellationDuration
     }
 
+    /// Adds a service to the group.
+    ///
+    /// If the group is currently running, the added service will be started immediately.
+    /// If the group is gracefully shutting down, cancelling, or already finished, the added service will not be started.
+    /// - Parameters:
+    ///   - serviceConfiguration: The service configuration to add.
+    public func addService(_ serviceConfiguration: ServiceGroupConfiguration.ServiceConfiguration) async {
+        switch self.state {
+        case var .initial(services: services):
+            self.state = .initial(services: [])
+            services.append(serviceConfiguration)
+            self.state = .initial(services: services)
+
+        case .running(_, let addedServiceChannel):
+            await addedServiceChannel.send(serviceConfiguration)
+
+        case .finished:
+            return
```

Comment:
I think this case definitely needs to throw an error to indicate something went wrong

Reply:
I think that's fair. Let's ignore my comment then. `addService` is a best effort thing and if the group is finished or shutting down it won't run the new service. I think we just need to add a debug log for when that happens.